### PR TITLE
merge the common-lib module and the google-cloud-core modules as they…

### PR DESCRIPTION
**Pull request by patflynn**
… serve the same purpose.

### CODE REVIEW

File: common-test-lib/build.gradle [hunk 1]
```diff
@@ -15,13 +15,13 @@
  */
 
 dependencies {
+    compile project(':google-cloud-tools-plugin:google-cloud-core')
```
Comment (ivanporty) [inline common-test-lib/build.gradle]: It looks like `common-test-lib` depends on `core` now, and `core` tests depend on `common-test-lib` in their turn. I can't remember off the top of my head if this is handled well by Gradle, what do you think, could this eventually lead to circular issues when compiling?

**Comment by patflynn** [comment]
I think this isn't a problem as only the tests in core depend on
common-test-lib.

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java [hunk 2]
```diff
@@ -76,21 +76,24 @@ public OSProcessHandler createProcessHandler(
             AppEngineServerModel runConfiguration;
 
             try {
-              // Getting the clone so the debug flags aren't added to the persisted settings.
+              // Getting the clone so the debug com.google.cloud.tools.intellij.flags aren't added
```
Comment (etanshaul) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: comment refactor accident?

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java [hunk 3]
```diff
@@ -76,21 +76,24 @@ public OSProcessHandler createProcessHandler(
...
+              // to the persisted settings.
               runConfiguration = (AppEngineServerModel) commonModel.getServerModel().clone();
             } catch (CloneNotSupportedException ee) {
               throw new ExecutionException(ee);
             }
 
             Map<String, String> environment = Maps.newHashMap(configuredEnvironment);
 
-            // IntelliJ appends the JVM flags to the environment variables, keyed by an empty
+            // IntelliJ appends the JVM com.google.cloud.tools.intellij.flags to the environment
```
Comment (etanshaul) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: same here

Comment (patflynn) [reply google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: yep! fixed.

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java (hunk 2 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: Lol, I thought this is intentional clarification on which flags are used

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java [hunk 4]
```diff
@@ -76,21 +76,24 @@ public OSProcessHandler createProcessHandler(
...
+            // variables, keyed by an empty
             // string; so we need extract them here.
             String jvmFlags = environment.get(JVM_FLAGS_ENVIRONMENT_KEY);
             if (jvmFlags != null) {
               runConfiguration.appendJvmFlags(Arrays.asList(jvmFlags.trim().split(" ")));
             }
-            // We don't want to pass the jvm flags to the dev server environment
+            // We don't want to pass the jvm flags to the dev server
+            // environment
```
Comment (etanshaul) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: nit: does this need to be on a new line?

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java (hunk 3 above)
Comment (etanshaul) [reply google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: its still showing: com.google.cloud.tools.intellij.flags. Forgot to push maybe?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/migration/StagedArtifactNameConverterProvider.java [hunk 5]
```diff
@@ -43,9 +43,9 @@
  * the built artifact to "target.jar" or "target.war", depending on the type of artifact that was
  * built.
  *
- * <p>This conversion flags all deployment configurations with a legacy bit that will be checked
- * upon deployment. If set, the staged artifact name will be explicitly set to "target.jar" or
- * "target.war" so the functionality is preserved.
+ * <p>This conversion com.google.cloud.tools.intellij.flags all deployment configurations with a
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/migration/StagedArtifactNameConverterProvider.java]: same comment refactoring problem (flags) here too

Comment (etanshaul) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/migration/StagedArtifactNameConverterProvider.java]: there are several more here that I missed. perhaps do a global search?

**Comment by patflynn** [comment]
OK tried again. PTAL.

**Comment by ivanporty** [comment]
Looking a bit further for integration, would it possibly make sense to also integrate `common-test-lib` into `core` as part of its test dependencies and code? Or we have a specific reasons to keep it separate?

**Comment by etanshaul** [comment]
good point -now that the account plugin is integrated. I suppose this could be done too in another PR?

**Comment by patflynn** [comment]
It's something to discuss.  Usually you don't want to package your
test-infra code into your binary so it's kind of nice to have as a separate
module.  This does add project complexity though so I'm open to bundling it
into core.  Definitely not part of this PR though.

**Comment by etanshaul** [comment]
thats true - we eventually could have new integration / ui test modules that depend on this common test infra. so its nice to have it separate like you said.

**Comment by patflynn** [comment]
@etanshaul.  Yeah. Ideally we'd break the link from test-lib to core.  It's only for ThreadUtil

**Pull request merged**